internal/pkg/service/buffer/sink/tablesink/storage/repository/slice_repo.go

```diff
@@ -0,0 +1,31 @@
+
+// newSlice creates slice definition.
+func newSlice(now time.Time, file storage.File, volumeID storage.VolumeID, prevSliceSize datasize.ByteSize) (s storage.Slice, err error) {
+	// Validate compression type.
+	// Other parts of the system are also prepared for other types of compression,
+	// but now only GZIP is supported in the Keboola platform.
+	switch file.LocalStorage.Compression.Type {
+	case compression.TypeNone, compression.TypeGZIP: // ok
+	default:
+		return storage.Slice{}, errors.Errorf(`file compression type "%s" is not supported`, file.LocalStorage.Compression.Type)
+	}
```

Comment:
If you mean this:

Condition 1:
![image](https://github.com/keboola/keboola-as-code/assets/19371734/2561e0ad-80b1-4e40-a8c9-c9788c576d52)

Condition 2:
![image](https://github.com/keboola/keboola-as-code/assets/19371734/8d4b2bb7-ee29-488e-af5a-c0c4d19a3c97)


-----

**Then yes, it is necessary, or better word will be: it is intentionally.**

Condition 1 tells:
- Keboola STAGING storage support None and GZIP compression.

Condition 2 tells:
- We currently only use None and Gzip compression at all (because other combinations did not make sense in the benchmarks).

-----------------

There are three storage level and each can have a compression:
- Local - local disk - implemented: `None`, `ZSTD`, `GZIP`
- Staging - AWS S3/Azure Blob Storage - supported: `None`, `Gzip`.
- Target - Snowflake internal comression (out of the Stream API scope)

So possible combinations are ([implemented here](https://github.com/keboola/keboola-as-code/blob/feddec257f1f9534ae424b9209c7aec482f792f1/internal/pkg/service/buffer/sink/tablesink/storage/level/local/reader/volume/reader.go#L77-L121), [tested here](https://github.com/keboola/keboola-as-code/blob/feddec257f1f9534ae424b9209c7aec482f792f1/internal/pkg/service/buffer/sink/tablesink/storage/level/local/reader/volume/reader_test.go#L87-L117)):
- :heavy_check_mark: Local `None` -> Staging `None` -> Target 
  - :arrow_up: disk IO; :arrow_up: network IO; :arrow_down: CPU
- :heavy_check_mark: Local `None` -> Staging `Gzip` -> Target 
  -  :arrow_up: disk IO; :arrow_down: network IO; :arrow_up: CPU
- :heavy_check_mark: Local `Gzip` -> Staging `Gzip` -> Target 
  -  :arrow_down: disk IO; :arrow_down: network IO; :arrow_up: CPU
- :x: Local `ZSTD` -> Staging `Gzip` -> Target 
  -  :arrow_down: disk IO; :arrow_down: network IO; :arrow_up::arrow_up: CPU
  
The last combination is implemented, but banned.

But it may change in the future.
- For example we will find that Brotli -> GZIP makes sense under some circumstances.

BTW: 
- It is not mainly about the fact that decompressing something and re-compressing it again, in different format, is inefficient.
- The main benefit is that when we **send a file from the disk to the network without changes**, we can use the Direct Memory Access (DMA) transfer between the disk and the network, outside of the application.
- Read more: https://itnext.io/optimizing-large-file-transfers-in-linux-with-go-an-exploration-of-tcp-and-syscall-ebe1b93fb72f